crates/cairo-addons/src/vm/dict_manager.rs

```diff
@@ -82,6 +85,51 @@ impl PyTrackerMapping {
     }
 }
 
+// Object returned by DictManager.preimages enabling access to the preimages by index and mutating
+/// the preimages with manager.preimages[index] = preimage
+#[pyclass(name = "PreimagesMapping", unsendable)]
+pub struct PyPreimagesMapping {
+    inner: Rc<RefCell<RustDictManager>>,
```

Comment:
Why are we manipulating this object?

Reply:
the preimages are a field of the RustDictManager